index.js

```diff
@@ -210,7 +210,26 @@ Filter.prototype.canProcessFile =
   return !!this.getDestFilePath(relativePath);
 };
 
+Filter.prototype.isDirectory = function(relativePath) {
+  if (this.inputPaths === undefined) {
+    return false;
+  }
+
+  var srcDir = this.inputPaths[0];
+  var path = srcDir + '/' + relativePath;
+
+  return fs.lstatSync(path).isDirectory();
```

Comment:
rather then stating again, lets pass `entry` from https://github.com/stefanpenner/broccoli-persistent-filter/blob/master/index.js#L125 to `getDestFilePath`, that way we can simply ask it `entry.isDirectory()` etc.